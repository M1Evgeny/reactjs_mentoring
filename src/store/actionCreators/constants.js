export const constants = {
  BASE_URL: "http://localhost:4000",
};
