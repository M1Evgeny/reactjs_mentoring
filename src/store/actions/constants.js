export const constants = {
  FETCH_MOVIES_REQUEST: "FETCH_MOVIES_REQUEST",
  FETCH_MOVIES_SUCCESS: "FETCH_MOVIES_SUCCESS",
  FETCH_MOVIES_FAILURE: "FETCH_MOVIES_FAILURE",
  SORT_MOVIES: "SORT_MOVIES",
  FILTER_MOVIES: "FILTER_MOVIES",
  FETCH_MOVIE_BY_ID_REQUEST: "FETCH_MOVIE_BY_ID_REQUEST",
  SET_MOVIE_ID: "SET_MOVIE_ID"
};
