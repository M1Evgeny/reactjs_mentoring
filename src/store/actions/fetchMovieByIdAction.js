import { constants } from "./constants";

export const fetchMovieSuccess = (data) => {
  return {
    type: constants.FETCH_MOVIE_BY_ID_REQUEST,
    payload: data,
  };
};
