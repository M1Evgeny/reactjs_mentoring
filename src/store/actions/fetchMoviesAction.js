import { constants } from "./constants";

const fetchFilmsSuccess = (data) => {
  return {
    type: constants.FETCH_MOVIES_SUCCESS,
    payload: data,
  };
};

const fetchMoviesRequest = () => {
  return {
    type: constants.FETCH_MOVIES_REQUEST,
  };
};

const fetchMoviesFailure = (error) => {
  return {
    type: constants.FETCH_MOVIES_FAILURE,
    payload: error.message,
  };
};

export { fetchFilmsSuccess, fetchMoviesRequest, fetchMoviesFailure };
