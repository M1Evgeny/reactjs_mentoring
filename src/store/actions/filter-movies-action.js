import { constants } from "./constants";

export const movieGenre = (genre) => {
  return {
    type: constants.FILTER_MOVIES,
    payload: genre,
  };
};
