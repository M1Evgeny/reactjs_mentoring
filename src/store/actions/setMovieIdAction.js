import { constants } from "./constants";

export const setMovieId = (data) => ({
  type: constants.SET_MOVIE_ID,
  payload: data,
});
