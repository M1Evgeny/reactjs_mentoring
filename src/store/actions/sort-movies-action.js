import { constants } from "./constants";

export const sort = (sortParam) => ({
  type: constants.SORT_MOVIES,
  payload: sortParam,
});
