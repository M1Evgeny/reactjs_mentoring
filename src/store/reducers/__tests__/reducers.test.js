import { movieReducer } from "..";
import { constants } from "../../actions/constants";

describe(">>>R E D U C E R --- Test movieReducer", () => {
  it("+++ reducer with shapshot", () => {
    let state = {};
    expect(movieReducer(undefined, { type: "default" })).toMatchSnapshot();
  });

  it("+++ reducer for FETCH_MOVIES_REQUEST", () => {
    let state = { loading: false };
    state = movieReducer(state, { type: constants.FETCH_MOVIES_REQUEST });
    expect(state).toEqual({ loading: true });
  });

  it("+++ reducer for FETCH_MOVIES_SUCCESS", () => {
    const films = [1, 2, 3];
    let state = { films: [] };
    state = movieReducer(state, {
      type: constants.FETCH_MOVIES_SUCCESS,
      payload: films,
    });
    expect(state).toEqual({ films: films, error: "", loading: false });
  });

  it("+++ reducer for FETCH_MOVIES_FAILURE", () => {
    let state = {};
    state = movieReducer(state, {
      type: constants.FETCH_MOVIES_FAILURE,
      payload: "test",
    });
    expect(state).toEqual({ loading: false, error: "test" });
  });

  it("+++ reducer for SORT_MOVIES", () => {
    let state = {};
    state = movieReducer(state, {
      type: constants.SORT_MOVIES,
      payload: "test",
    });
    expect(state).toEqual({ sortParam: "test" });
  });

  it("+++ reducer for FILTER_MOVIES", () => {
    let state = {};
    state = movieReducer(state, {
      type: constants.FILTER_MOVIES,
      payload: "test",
    });
    expect(state).toEqual({ genre: "test" });
  });

  it("+++ reducer for FETCH_MOVIE_BY_ID_REQUEST", () => {
    let state = {};
    state = movieReducer(state, {
      type: constants.FETCH_MOVIE_BY_ID_REQUEST,
      payload: "test",
    });
    expect(state).toEqual({ movie: "test" });
  });

  it("+++ reducer for SET_MOVIE_ID", () => {
    let state = {};
    state = movieReducer(state, {
      type: constants.SET_MOVIE_ID,
      payload: "test",
    });
    expect(state).toEqual({ movieId: "test" });
  });
});
