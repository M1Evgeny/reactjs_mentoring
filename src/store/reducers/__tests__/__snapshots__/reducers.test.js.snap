// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>R E D U C E R --- Test movieReducer +++ reducer with shapshot 1`] = `
Object {
  "error": "",
  "films": Array [],
  "loading": false,
  "movie": Object {
    "budget": "",
    "genres": Array [],
    "id": "",
    "overview": "",
    "poster_path": "",
    "release_date": "",
    "revenue": "",
    "runtime": "",
    "tagline": "",
    "title": "",
    "vote_average": "",
    "vote_count": "",
  },
  "movieId": "",
}
`;
